### PR TITLE
Update diagnostic usage for langversion in C#

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/17147

With this fix, invoking `csc /langversion:1234` displays:
```
Microsoft (R) Visual C# Compiler version 42.42.42.42424
Copyright (C) Microsoft Corporation. All rights reserved.

error CS1617: Invalid option '1234' for /langversion; must be ISO-1, ISO-2, Default or an integer in range 1 to 7.
warning CS2008: No source files specified.
error CS1562: Outputs without source must have the /out option specified
```

### CODE REVIEW

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for one-line message change for `csc`.
`vbc` does not have a corresponding message.

**Comment by jcouv** [comment]
@gafter @VSadov It occurred to me during standup that we could catch this in the future with a test. Added.

File: src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs [hunk 1]
```diff
@@ -1320,6 +1320,10 @@ public void LangVersion()
             parsedArgs = DefaultParse(new[] { "/langversion: ", "a.cs" }, _baseDirectory);
             parsedArgs.Errors.Verify(Diagnostic(ErrorCode.ERR_SwitchNeedsString).WithArguments("<text>", "/langversion:"));
             Assert.Equal(defaultVersion, parsedArgs.ParseOptions.LanguageVersion);
+
+            // When new language versions are added, this test will fail. Remember to update the diagnostics message (ERR_BadCompatMode).
+            Assert.Equal(LanguageVersion.CSharp7, LanguageVersion.Latest.MapSpecifiedToEffectiveVersion());
+            Assert.Equal(LanguageVersion.CSharp7, LanguageVersion.Default.MapSpecifiedToEffectiveVersion());
```
Comment (cston) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs]: Minor point: perhaps move these checks to a separate test.

Comment (jcouv) [inline src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs]: Sure. I split it out.

**Comment by jcouv** [comment]
Moving to master branch, since doesn't meet bar for 15.1.

**Comment by jcouv** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest please

// error MSB4018: The "FindInstalledExtension" task failed unexpectedly. [D:\j\workspace\windows_debug---b9840ae3\src\Samples\VisualBasic\ConvertToAutoProperty\Impl\ConvertToAutoPropertyVB.vbproj]

**Comment by jcouv** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest please

**Pull request merged**